WcaOnRails/app/views/shared/_associated_events_picker.html.erb

```diff
@@ -0,0 +1,36 @@
+<%#
+Variables:
+ - form_builder
+ - label
+ - disabled
+ - events_association_name -> name of the associated events relation
+ - allowed_events -> an Array of Event objects that should be shown (all by default)
+ - selected_events -> an Array of Event objects that whould be marked as selected
+%>
+
+<% label ||= "Events" %>
+<% disabled ||= false %>
+<% allowed_events ||= Event.all_official %>
+
+<%# Note: form_builder.object is the object having associated events. %>
+<% associated_events = form_builder.object.send(events_association_name)
+   all_events = allowed_events.map do |event|
+     associated_events.find_by_event_id(event.id) || associated_events.build(event_id: event.id)
+   end %>
```

Comment:
Actually I used this kinda naturally and it wasn't weird to me that it works well =)
What do you mean with this `end`? It just *end*s the block.